core/src/main/java/org/apache/iceberg/util/TableScanUtil.java

```diff
@@ -35,16 +38,22 @@
 import org.apache.iceberg.ScanTaskGroup;
 import org.apache.iceberg.SplittableScanTask;
 import org.apache.iceberg.StructLike;
+import org.apache.iceberg.io.CloseableGroup;
 import org.apache.iceberg.io.CloseableIterable;
+import org.apache.iceberg.io.CloseableIterator;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 import org.apache.iceberg.relocated.com.google.common.collect.FluentIterable;
 import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
 import org.apache.iceberg.relocated.com.google.common.collect.Iterables;
 import org.apache.iceberg.relocated.com.google.common.collect.Lists;
 import org.apache.iceberg.relocated.com.google.common.collect.Maps;
 import org.apache.iceberg.types.Types;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class TableScanUtil {
+  private static final Logger LOG = LoggerFactory.getLogger(TableScanUtil.class);
+  private static final long MIN_SPLIT_SIZE = 16 * 1024 * 1024; // 16 MB
```

Comment:
This feels like it's too large for a min split size. Either that or we should take into account what the row group size is actually configured to. For example, if the row group was set to 8MB, then we still wouldn't be able to achieve maximum parallelism because of this arbitrary default.

Reply:
What do you think it should be instead? 8 MB?

This seems pretty reasonable to me. My rationale is that this min size should balance task overhead. We want to process enough data that we don't waste time in overhead, but not so much that we have a delay because tasks are too large. 16 MB will be processed fairly quickly and I doubt that it would be a very noticeable bottleneck. On the other hand, our file open cost is 4 MB. I would say that we want to be higher than the file open cost since that's a rough measure of overhead. So 8 or 16 seems about right to me.

Reply:
I don't think we want to focus too much on optimization/cost of single tasks here.  The goal is to achieve better overall wall-clock performance through parallelism and sacrifice some individual task efficiency for overall throughput.  I would go as low as conceivably possible, which means either a value like this or the row group size if configured since that's the lowest splittable unit of work.

Reply:
So you're arguing that there should be no minimum size? Seems like we might get some silly cases, like a 64k target size. That should be mostly mitigated by Parquet, but we could have a small minimum for Avro, like 4MB.

Reply:
I would say 8 MB is already small enough so such tasks should proceed fairly quickly. I would be a bit concerned going smaller than that. The cost of opening files is non-trivial. Also, too small splits may overload the underlying storage with a large number of requests. I even find 16 MB reasonable, to be honest. I don't want our queries to fail with rate limits.

Reply:
+1 on 16 MB min size